libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -846,6 +846,18 @@ - (void)setupWithCredentials:(SFOAuthCredentials*) credentials {
     self.idCoordinator.delegate = self;
 }
 
+- (void)restartAuthentication {
+    @synchronized (self.authBlockList) {
+        if (!self.authenticating) {
+            [self log:SFLogLevelWarning format:@"%@: Authentication manager is not currently authenticating.  No action taken.", NSStringFromSelector(_cmd)];
+            return;
+        }
+        [self log:SFLogLevelInfo format:@"%@: Restarting in-progress authentication process.", NSStringFromSelector(_cmd)];
+        [self.coordinator stopAuthentication];
+        [self loginWithCredentials:[self createOAuthCredentials]];
```

Comment:
Passing in credentials is effectively synonymous with an auth refresh, which wouldn't be expected to happen in conjunction with this method being called.

I take your point about the idea of making this generic enough to restart any in-progress authentication.  But if we do that, we're either going to have to change this interface to provide the auth configuration that needs to change—in this case, the login host—or use existing methods and more or less break encapsulation by specifying all of that information directly to the existing credentials object from the login host picker, before calling into authentication again.

There may be another way or middle ground, but I haven't seen it yet.